WireDomain/Tests/WireDomainTests/LocalStores/ConversationLabelsLocalStoreTests.swift

```diff
@@ -0,0 +1,310 @@
+
+        try await context.perform { [self] in
+            let fetchRequest = NSFetchRequest<Label>(entityName: Label.entityName())
+            let results = try context.fetch(fetchRequest)
+            let label = try XCTUnwrap(results.first)
+            XCTAssertEqual(label.remoteIdentifier, Scaffolding.favoriteConversationLabel1.id)
+        }
+
+        // When
+
+        try await sut.deleteOldLabelsLocally(
+            excludedLabels: [Scaffolding.conversationLabel2, Scaffolding.conversationLabel3]
+        )
+
+        // Then
+
+        try await context.perform { [context] in
+            let fetchRequest = NSFetchRequest<Label>(entityName: Label.entityName())
+            let results = try context.fetch(fetchRequest)
+            let favoriteLabel = try XCTUnwrap(results.first)
+
+            XCTAssertEqual(existingFavoriteLabel, favoriteLabel) // favorite label was not removed
+        }
+    }
+
+}
+
+private extension ConversationLabelsLocalStoreTests {
+    func cleanUpEntity() async {
+        await context.perform { [self] in
+            let fetchRequest: NSFetchRequest<NSFetchRequestResult> = Label.fetchRequest()
+            let batchDeleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
+            _ = try? context.execute(batchDeleteRequest)
+        }
+    }
+
+    func setupConversations() async {
+        await context.perform { [self] in
+            conversation1 = ZMConversation.insertNewObject(in: context)
+            conversation1.remoteIdentifier = Scaffolding.conversationLabel1.conversationIDs[0]
+
+            conversation2 = ZMConversation.insertNewObject(in: context)
+            conversation2.remoteIdentifier = Scaffolding.conversationLabel1.conversationIDs[1]
+
+            conversation3 = ZMConversation.insertNewObject(in: context)
+            conversation3.remoteIdentifier = Scaffolding.updatedConversationLabel1.conversationIDs[2]
+        }
+    }
+
+    private enum Scaffolding {
+        static let conversationLabel1 = ConversationLabelInfo(
+            id: .mockID1,
+            name: "ConversationLabel1",
+            type: 0,
+            conversationIDs: [
+                .mockID2,
+                .mockID3
+            ]
+        )
+
+        static let updatedConversationLabel1 = ConversationLabelInfo(
+            id: .mockID1,
+            name: "UpdatedConversationLabel1", /// Updated name
+            type: 0,
+            conversationIDs: [
+                .mockID2,
+                .mockID3,
+                .mockID4 /// new conversation added
+            ]
+        )
+
+        static let favoriteConversationLabel1 = ConversationLabelInfo(
+            id: .mockID3,
+            name: "FavoriteConversationLabel1",
+            type: 1, /// this label is favorite
+            conversationIDs: [
+                .mockID1,
+                .mockID2
+            ]
+        )
+
+        static let conversationLabel2 = ConversationLabelInfo(
+            id: .mockID4,
```

Comment:
now se use UUID helpers from WireTestingPackage

Reply:
Do we specifically need predefined ids or would it work to create random ids?

Reply:
It would work creating random ids, I figured it would be nice to harmonize the code by using our UUID helper across all tests